pydantic/_internal/_model_construction.py

```diff
@@ -246,7 +246,6 @@ def _collect_bases_data(bases: tuple[type[Any], ...]) -> tuple[set[str], set[str
         'The `__fields__` attribute is deprecated, use `model_fields` instead.', category=PydanticDeprecatedSince20
     )
     def __fields__(self) -> dict[str, FieldInfo]:
-        warnings.warn('The `__fields__` attribute is deprecated, use `model_fields` instead.', DeprecationWarning)
```

Comment:
The reason we have this warning is actually because this is how PyCharm decides how to show the strikethrough for deprecated attributes:
<img width="308" alt="image" src="https://github.com/pydantic/pydantic/assets/35119617/4db90560-d8ed-450d-8608-c5985cb5647f">
If this line is removed, PyCharm stops showing the strikethrough on usage, even after updating `typing_extensions`.

If there are downsides to having both I'm happy to try to resolve it but I think showing the deprecated stuff as deprecated in PyCharm is a significant benefit, considering we are trying to advise people against using deprecated stuff anyway (so I don't mind terribly if it emits two warnings, unless there is some significant downside to that).

Reply:
Sure thing I can add back the warning for properties, and as a workaround set `category=None` in the decorator to avoid having the warning emitted twice.

However, this is most likely an issue with PyCharm. Do you happen to know if there's an already existing issue tracking this?

The recently defined typing spec [states properties should also be marked as deprecated](https://typing.readthedocs.io/en/latest/spec/directives.html#deprecated) by type checkers (see provided example).

Reply:
Yeah it's definitely a PyCharm issue — I found this in their issue tracker https://youtrack.jetbrains.com/issue/PY-61651/Deprecation-highlighting-with-PEP-702-deprecated-decorator and added a comment on it about our usage within Pydantic.

Reply:
@dmontagu, is this also the case for standard methods/functions, or is it only the not working for properties?

Reply:
I think for everything not just properties

Reply:
Ok, then I will add back the actual `warn` calls, and use `category=None` on the decorator. I can also create an issue to remind us to remove them again once PyCharm will have implemented support for the decorator.

Reply:
@Viicos,

That would be fantastic if you could create an issue as a reminder for us to remove the `warn` calls

Reply:
Sure thing: https://github.com/pydantic/pydantic/issues/8524